diff --git a/node_modules/@types/orbit-db/Cache.d.ts b/node_modules/@types/orbit-db/Cache.d.ts
index 0dea665..c003fca 100644
--- a/node_modules/@types/orbit-db/Cache.d.ts
+++ b/node_modules/@types/orbit-db/Cache.d.ts
@@ -2,24 +2,22 @@ declare module "orbit-db-cache" {
   import KeyValueStore from "orbit-db-kvstore";
 
   export default class Cache<V> {
-      readonly _store: KeyValueStore<V>;
+    readonly _store?: KeyValueStore<V>;
 
-      constructor(store: KeyValueStore<V>)
+    constructor(store: KeyValueStore<V>)
 
 
-      status(): any;
+    close(): Promise<void>;
+    open(): Promise<void>;
 
-      close(): Promise<void>;
-      open(): Promise<void>;
 
+    get(key: string): Promise<V>;
 
-      get(key: string): Promise<V>;
+    set(key: string, value: V): Promise<void>;
+    put(key: string, value: V): Promise<void>;
+    del(key: string): Promise<void>;
 
-      set(key: string, value: V): Promise<void>;
-      put(key: string, value: V): Promise<void>;
-      del(key: string): Promise<void>;
-    
-      load(): void;
-      destroy(): void;
+    load(): void;
+    destroy(): void;
   }
 }
\ No newline at end of file
