diff --git a/node_modules/@types/orbit-db/Keystore.d.ts b/node_modules/@types/orbit-db/Keystore.d.ts
index 1bc5dd8..b2c8d62 100644
--- a/node_modules/@types/orbit-db/Keystore.d.ts
+++ b/node_modules/@types/orbit-db/Keystore.d.ts
@@ -1,5 +1,34 @@
 declare module "orbit-db-keystore" {
+    export interface IOrbitDBKeystoreStore {
+        status: string;
+        open(): Promise<void>;
+        close(): Promise<void>;
+        put(k: string, v :any): Promise<void>;
+        get(k: string): string | Uint8Array;
+    }
+
+    export interface IOrbitDBKeystoreCache {
+        length: number;
+        keys: string[];
+        set(k: string, v :any): void;
+        get(k: string): any;
+        remove(k: string): void;
+        clear(): void;
+    }
+    
+    export interface IOrbitDBKeystoreOptionsForSecretStorage {
+        store?: string | IOrbitDBKeystoreStore;
+        cache?: IOrbitDBKeystoreCache;
+        dbName?: string;
+        credentials: {
+            password: string;
+        }
+    }
+
+   export type TOrbitDBKeystoreOptions = IOrbitDBKeystoreOptionsForSecretStorage | string;
+    
     export class Keystore {
+        constructor(options: TOrbitDBKeystoreOptions);        
 
         hasKey(id:string): Promise<boolean>
 
